public/app/plugins/datasource/jaeger/datasource.ts

```diff
@@ -146,3 +126,62 @@ function getTime(date: string | DateTime, roundUp: boolean) {
   }
   return date.valueOf() * 1000;
 }
+
+function createTraceFrame(data: any): DataFrame {
+  return new MutableDataFrame({
+    fields: [
+      {
+        name: 'trace',
+        type: FieldType.trace,
+        values: data,
+      },
+    ],
+    meta: {
+      preferredVisualisationType: 'trace',
+    },
+  });
+}
+
+function createGraphFrames(data: TraceViewData[]): DataFrame[] {
+  const { nodes, edges } = convertTraceToGraph(data[0]);
+
+  return [
+    new MutableDataFrame({
+      name: 'nodes',
+      fields: mapFields(nodes),
+      meta: { preferredVisualisationType: 'nodeGraph' },
+    }),
+
+    new MutableDataFrame({
+      name: 'edges',
+      fields: mapFields(edges),
+      meta: { preferredVisualisationType: 'nodeGraph' },
+    }),
+  ];
+}
+
+function mapFields(data: Array<Record<string, any>>): FieldDTO[] {
+  const map = data.reduce((acc, datum) => {
+    for (const key of Object.keys(datum)) {
+      if (!acc[key]) {
+        if (key === Fields.color) {
+          acc[key] = createField(key, { color: { mode: 'continuous-GrYlRd' } });
```

Comment:
So I wonder if this makes sense. Basically I am saying that with the node data you can also send a `color` column to say what color should the node have. But also you can specify a color more and then use it's numeric value to get final color. This means I had to change the type of `field.color.mode` because the enum type does not contain all the modes from `fieldColorModeRegistry` and as that registry is dynamic it probably can't. I assume this makes sense from how fields work and should be easy to do overrides later on in panel for example (so this is sort of default mainly for Explore set by the data source) @torkelo @dprokop